src/osd/PrimaryLogPG.cc

```diff
@@ -6665,6 +6665,11 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  result = op_finisher->execute();
 	  assert(result == 0);
 
+	  if (oi.manifest.redirect_target != hobject_t()) {
+	    object_locator_t target_oloc(oi.manifest.redirect_target);
+	    refcount_manifest(ctx->obc, target_oloc, oi.manifest.redirect_target, 
+			      SnapContext(), "chunk_put", NULL, 0);
+	  }
```

Comment:
I think this has to happen in a context *after* the local operation commits.  otherwise we might crash, failing to commit locally, but still decrement the ref on the chunk.

Reply:
Right. I will fix this as your comment.

Reply:
It seems like this shouldn't be based on need_reference but instead on just (oi.flags & object_info_t::FLAG_REDIRECT_HAS_REFERENCE)